File: java/src/network/lobby/packets/user/LockUserPacket.java
package network.lobby.packets.user;

import misc.compress.*;
import misc.Debug;
import network.lobby.packets.UserPacket;

public class LockUserPacket extends UserPacket
{
	private boolean locked;

	public LockUserPacket(boolean value)
	{
		locked = value;
	}

	public void assign(LockUserPacket lockPacket)
	{
		Debug.warnIf(lockPacket == null, "LockUserPacket.assign(null)");
		locked = lockPacket.locked;
	}

	public LockUserPacket(LockUserPacket packet)
	{
		locked = packet.isLocked();
	}

	public boolean isLocked()
	{
		return locked;
	}

	public LockUserPacket(CompressBuffer buffer)
	{
		locked = buffer.cutBoolean();
	}

	@Override public byte getCID() { return Compressor.LOCK_USER_PACKET_CID; }
	@Override public byte[] compress()
	{
		return Compressor.compressBoolean(locked);
	}
}

File: cpp/src/network/packets/LockUserPacket.cpp
#include "LockUserPacket.hpp"

#include <misc/compress/CompressBuffer.hpp>

LockUserPacket::LockUserPacket(bool locked) : locked(locked)
{}

LockUserPacket::LockUserPacket(CompressBuffer* buffer)
	: locked(buffer->cutBool())
{}

bool LockUserPacket::isLocked() const
{
	return locked;
}

std::string LockUserPacket::getCompressString() const
{
	return compressBool(isLocked());
}

CompressID LockUserPacket::getCompressID() const
{
	return LOCK_USER_PACKET_CID;
}
